lib/config-generator.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * @fileoverview to generate config files.
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+import process from "process";
+import path from "path";
+import { spawnSync } from "child_process";
+import { writeFile } from "fs/promises";
+import enquirer from "enquirer";
+import { isPackageTypeModule, installSyncSaveDev, fetchPeerDependencies, findPackageJson } from "./utils/npm-utils.js";
+import { getShorthandName } from "./utils/naming.js";
+import * as log from "./utils/logging.js";
+
+// TODO: need to specify the package version - they may export flat configs in the future.
+const jsStyleGuides = [
+    { message: "Airbnb: https://github.com/airbnb/javascript", name: "airbnb", value: { packageName: "eslint-config-airbnb", type: "eslintrc" } },
+    { message: "Standard: https://github.com/standard/standard", name: "standard", value: { packageName: "eslint-config-standard", type: "eslintrc" } },
+    { message: "XO: https://github.com/xojs/eslint-config-xo", name: "xo", value: { packageName: "eslint-config-xo", type: "eslintrc" } }
+];
+const tsStyleGuides = [
+    { message: "Standard: https://github.com/standard/eslint-config-standard-with-typescript", name: "standard", value: { packageName: "eslint-config-standard-with-typescript", type: "eslintrc" } },
+    { message: "XO: https://github.com/xojs/eslint-config-xo-typescript", name: "xo", value: { packageName: "eslint-config-xo-typescript", type: "eslintrc" } }
+];
+
+/**
+ * Class representing a ConfigGenerator.
+ */
+export class ConfigGenerator {
+
+    /**
+     * Create a ConfigGenerator.
+     * @param {Object} options The options for the ConfigGenerator.
+     * @param {string} options.cwd The current working directory.
+     * @param {Object} options.answers The answers provided by the user.
+     * @returns {ConfigGenerator} The ConfigGenerator instance.
+     */
+    constructor(options = {}) {
+        this.cwd = options.cwd || process.cwd();
```

Comment:
It looks like `options.cwd` will be always set, so maybe we don't need the fallbacks?

```suggestion
    constructor(options) {
        this.cwd = options.cwd;
```